### PR TITLE
Define e2e execution logic in ols repo

### PR DESCRIPTION
**Pull request by bparees**
## Description
Defines a script that generates a config file for OLS, starts the server, and
then runs our e2e tests against the OLS server.

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Related Tickets & Documents

- Related Issue #
- Closes #

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

### CODE REVIEW

**Comment by bparees** [comment]
@tisnik ptal

once this merges i'll update our current CI e2e job to invoke this script.

File: scripts/test-e2e-standalone.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+# 3) Invoke the test-e2e Makefile target
+# 4) Terminate the OLS server
+
+# temp directory for config file, output logs
+TMPDIR=$(mktemp -d)
+
+# ARTIFACT_DIR is defined when running in a prow job
+ARTIFACT_DIR=${ARTIFACT_DIR:-$TMPDIR}
+
+export OLS_CONFIG_FILE=$ARTIFACT_DIR/olsconfig.yaml
```
Comment (tisnik) [inline scripts/test-e2e-standalone.sh:19]: double quotes to prevent globbing and splitting

File: scripts/test-e2e-standalone.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+OLS_LOGS=$ARTIFACT_DIR/ols.log
+
+# Generate ols config.yaml
+export PROVIDER=${PROVIDER:-openai}
```
Comment (tisnik) [inline scripts/test-e2e-standalone.sh:23]: double quotes to prevent globbing and splitting

File: scripts/test-e2e-standalone.sh [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+export PROVIDER_KEY_PATH=${PROVIDER_KEY_PATH:-/home/bparees/creds/llms/openai.key}
+export MODEL=${MODEL:-gpt-3.5-turbo-1106}
```
Comment (tisnik) [inline scripts/test-e2e-standalone.sh:25]: double quotes to prevent globbing and splitting

File: scripts/test-e2e-standalone.sh (hunk 2 above)
Comment (tisnik) [reply scripts/test-e2e-standalone.sh:23]: (it's a bit theoretical, but probably better to be on safe side)

File: scripts/test-e2e-standalone.sh (hunk 3 above)
Comment (tisnik) [reply scripts/test-e2e-standalone.sh:25]: (it's a bit theoretical, but probably better to be on safe side)

File: scripts/test-e2e-standalone.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+envsubst < `pwd`/tests/config/singleprovider.e2e.template.config.yaml > $OLS_CONFIG_FILE
```
Comment (tisnik) [inline scripts/test-e2e-standalone.sh:26]: now people tend to use `$()` instead of backticks

File: scripts/test-e2e-standalone.sh [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+echo "Installing dependencies"
+make install-deps && make install-deps-test
+
+echo Starting OLS server
+make run >& $OLS_LOGS &
+function finish {
```
Comment (tisnik) [inline scripts/test-e2e-standalone.sh:33]: IMHO it is not POSIX compliant. `finish()` is (yes, it's less readable)

**Pull request merged**